File: 150. Evaluate Reverse Polish Notation.go
func evalRPN(tokens []string) int {
    res := []int{}
    var result int
    operations := map[string]bool {
        "*": true,
        "+": true,
        "-": true,
        "/": true,
    }
    for _, token := range(tokens) {
        if operations[token] && len(res) > 0 {
            a := res[len(res)-1]
            b := res[len(res)-2]
            if token == "+"{
                result = a+b
            }else if  token == "-" {
                result = b-a
            }else if  token == "*" {
                result = b*a
            }else if  token == "/" {
                result = int(b/a)
            }
            res = res[:len(res)-2]
            res = append(res, result)
        } else {
            num, _ := strconv.Atoi(token)
            res = append(res, num)
        }
    }
    return res[0]
    
}

File: 150. Evaluate Reverse Polish Notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        res = []
        ops = ["+","-","*","/"]
        for token in tokens:
            if token in ops and res:
                a = res.pop()
                b = res.pop()
                if token == ops[0]:
                    res.append(a+b)
                if token == ops[1]:
                    res.append(b-a)
                if token == ops[2]:
                    res.append(a*b)
                if token == ops[3]:
                    res.append(int(b/a))
            else:
                res.append(int(token))
        return int(res[0])
        